### ISSUE TITLE
colour contrast issues reported on visually hidden elements

### ISSUE THREAD

**Issue by malcomio** [pa11y/pa11y#311]
running pa11y against elements that have been visually hidden using Jonathan Snook's clip method will report issues with their colour contrast ratio.

This would seem to be a false positive.

```
// Makes an element visually hidden, but accessible.
// @see http://snook.ca/archives/html_and_css/hiding-content-for-accessibility
@mixin element-invisible {
  position: absolute !important;
  height: 1px;
  width: 1px;
  overflow: hidden;
  clip: rect(1px, 1px, 1px, 1px);
}
```

Here's the actual output from `pa11y https://capgemini.github.io/categories --ignore "warning;notice"`

```
 • Error: This element has insufficient contrast at this conformance level. Expected a contrast ratio of at least 4.5:1, but text in this element has a contrast ratio of 2.48:1. Recommendation: change text colour to #757575.
   ├── WCAG2AA.Principle1.Guideline1_4.1_4_3.G18.Fail
   ├── #notepad-page-container > header > div:nth-child(1) > div > ul > li:nth-child(8) > form > label
   └── <label for="search-terms">Search</label>
```

**Comment by rowanmanning** [comment]
Thanks @malcomio, we're aware of this and it's an issue that's been fixed in the latest version of [HTML CodeSniffer](https://github.com/squizlabs/HTML_CodeSniffer) (which Pa11y uses under the hood).

We updated HTML CodeSniffer recently but it introduced breaking builds for a few people so we had to revert. We'll be working on getting this update in as soon as possible.

In the meantime you can reference the latest version of HTML CodeSniffer yourself:

```
pa11y https://capgemini.github.io/categories --ignore "warning;notice" --htmlcs http://squizlabs.github.io/HTML_CodeSniffer/build/HTMLCS.js
```

**Comment by malcomio** [comment]
thanks - that's very helpful
